### PR TITLE
Prepare for translation tweaks

### PR DESCRIPTION
**Pull request by microbit-grace**
- Translate “Untitled” project names
- Translate icon names
- Translate Led icon picker aria labels

The top part of the this list of translation TODOs doc (private): https://paper.dropbox.com/doc/CreateAI-translation-thoroughness-review-ui.en.json--CZ97k~y_kLdxH~ffwjgPpsJEAQ-nRUKbyPBmyCsUY3L3YtNa

Filenames have not been translated as they're not as essential

### CODE REVIEW

File: src/hooks/project-hooks.tsx [hunk 1]
```diff
@@ -99,6 +111,18 @@ export const ProjectProvider = ({
...
+  const projectName = useStore((s) => s.project.header?.name);
+  const setProjectName = useStore((s) => s.setProjectName);
+  const defaultProjectName = useDefaultProjectName();
+
+  // If project name is null, set project name as translated "untitled" name.
+  useEffect(() => {
+    if (projectName === null) {
+      setProjectName(defaultProjectName);
+    }
+  }, [defaultProjectName, intl, projectName, setProjectName]);
```
Comment (microbit-grace) [inline src/hooks/project-hooks.tsx:124]: Not sure if this is the best way of doing it. The store can't access the translated untitled string

Comment (microbit-matt-hillsdon) [reply src/hooks/project-hooks.tsx:124]: Hmm this has ended up a bit unpleasant. Is it feasible to pass it in? I guess initialisation might be awkward - store init would need to be triggered from in a React context and we don't have a notion of waiting for init right now?

Comment (microbit-matt-hillsdon) [reply src/hooks/project-hooks.tsx:124]: Perhaps OK to use English "Untitled" or another placeholder and swap at the point of asking for a name for the project?

File: lang/ui.en.json [hunk 2]
```diff
@@ -411,6 +411,10 @@
     "defaultMessage": "Short collections of movement data lasting 1 second. Your data samples are only stored on your computer, they do not get sent to anyone else.",
     "description": "Tooltip for data samples heading"
   },
+  "default-project-name": {
+    "defaultMessage": "Translated Untitled",
```
Comment (microbit-grace) [inline lang/ui.en.json:415]: To be changed back to Untitled once ensured working

Comment (microbit-grace) [reply lang/ui.en.json]: [71b52fa](https://github.com/microbit-foundation/ml-trainer/pull/462/commits/71b52fa8bf965b9f5e2159fe6850dfd19ebd0633)

File: src/hooks/project-hooks.tsx (hunk 1 above)
Comment (microbit-grace) [reply src/hooks/project-hooks.tsx]: I think delaying the switch between using the translated untitled version makes sense 👍 I have made the change in this commit [f90f00d](https://github.com/microbit-foundation/ml-trainer/pull/462/commits/f90f00dd80af591e9e8f144be2aabcc43c91ae11) and have had follow-up commits to clean up/fix/revert other bits

File: src/components/CodeViewDefaultBlock.tsx [hunk 3]
```diff
@@ -12,6 +12,9 @@ const CodeViewDefaultBlock = ({
 }: CodeViewDefaultBlockProps) => {
   const intl = useIntl();
   const ledPattern = makecodeIcons[icon];
+  const iconName = intl.formatMessage({
+    id: `led-icon-option-${icon.toLocaleLowerCase()}`,
```
Comment (microbit-matt-hillsdon) [inline src/components/CodeViewDefaultBlock.tsx:16]: I doubt we want locale-specific lowercasing here.

File: src/hooks/project-hooks.tsx [hunk 4]
```diff
@@ -85,6 +85,26 @@ interface ProjectProviderProps {
   children: ReactNode;
 }
 
+const useDefaultProjectName = (): string => {
+  const intl = useIntl();
+  return intl.formatMessage({ id: "default-project-name" });
+};
+
+export const useHasUntitledProjectName = (): boolean => {
```
Comment (microbit-matt-hillsdon) [inline src/hooks/project-hooks.tsx:93]: Feels a bit of a mouthful. Maybe useProjectIsUntitled ?

File: src/components/CodeViewDefaultBlock.tsx (hunk 3 above)
Comment (microbit-grace) [reply src/components/CodeViewDefaultBlock.tsx]: [0862e4b](https://github.com/microbit-foundation/ml-trainer/pull/462/commits/0862e4b2ef83ada6ebbd7a0142147fb3e1490077)

File: src/hooks/project-hooks.tsx (hunk 4 above)
Comment (microbit-grace) [reply src/hooks/project-hooks.tsx]: [337f1d7](https://github.com/microbit-foundation/ml-trainer/pull/462/commits/337f1d783778d99331ac2f97a692170f490bbc44)

File: src/hooks/project-hooks.tsx (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/hooks/project-hooks.tsx]: I'd still like to play with this some more and see if there's a simpler option - but if I don't get to it today we'll go ahead with this.

Comment (microbit-matt-hillsdon) [reply src/hooks/project-hooks.tsx]: Had a play, can't see a reasonable-effort approach to improving this. Sorry for the pointless delay.

**Review by microbit-matt-hillsdon — APPROVED** [review]

File: src/components/LedIconPicker.tsx [hunk 5]
```diff
@@ -46,7 +48,14 @@ const LedIconPicker = ({ onIconSelected }: LedIconPicker) => {
                 {Object.keys(makecodeIcons).map((icon, idx) => (
                   <IconButton
                     key={idx}
-                    aria-label={`Select ${icon} icon`}
+                    aria-label={intl.formatMessage(
+                      { id: "select-icon-option-action-aria" },
+                      {
+                        iconName: intl.formatMessage({
+                          id: `led-icon-option-${icon.toLocaleLowerCase()}`,
```
Comment (microbit-matt-hillsdon) [inline src/components/LedIconPicker.tsx:55]:
```suggestion
                          id: `led-icon-option-${icon.toLowerCase()}`,
```

**Comment by microbit-matt-hillsdon** [comment]
Snuck in one more toLocaleLowerCase -> toLowerCase tweak.

**Comment by microbit-matt-hillsdon** [comment]
Will wait for build then merge.

**Pull request merged**